### ISSUE TITLE
allow warning on duplicate merge key

### ISSUE THREAD

**Issue by juliantaylor** [adrienverge/yamllint#561]
yamllint does not warn on duplicate merge keys:

```
main:
  <<: *anchor_one
  <<: *anchor_two
```

This should be done by merging a sequence like this:

    <<<: [*anchor_one, *anchor_two]

https://yaml.org/type/merge.html

This has a well defined ordering and you avoid differences due to parsers handling duplicates differently (basically for the same reason yamllint complains about duplicates in the first place).

Ignoring of duplicate merge keys was added intentionally https://github.com/adrienverge/yamllint/issues/88 so as its probably not uncommon to have this in files, maybe this warning should only be emitted when asked for.

**Comment by adrienverge** [comment]
Hello,

I understand that the duplication of merge keys inside a mapping is not allowed by the YAML spec, although some YAML parsers support it. Is that right?

If yes, it would be hard to re-enable a warning for this, now that many people use yamllint in production every day. Instead, we could use a new option so that anyone can choose to allow / forbid it:
```yaml
rules:
  key-duplicates:
    forbid-duplicated-merge-keys: true
```
What do you think?

**Comment by ericqt** [comment]
Would the change be somewhere here? https://github.com/adrienverge/yamllint/commit/1b379628d77cf7ee29a1c8c70be6b58f97a45664

**Comment by travbale** [comment]
@adrienverge 

I created a PR based on the suggestion you made about having a `forbid-duplicated-merge-keys`. When enabled (off by default given some parsers support it) it will flag multiple `<<` keys. Having this would be very helpful for parsers that do not support multiple merge keys in a mapping such as [Go's yaml package](https://github.com/go-yaml/yaml).

Additionally [RedHat's YAML LSP](https://github.com/redhat-developer/yaml-language-server) flags multiple merge keys in a mapping so I believe there is a good case to support this. I do understand not making it default on given the flexibility of offered by  parsers such as PyYAML.
